app/src/main/java/com/door43/translationstudio/newui/newtranslation/NewTargetTranslationActivity.java

```diff
@@ -153,12 +160,24 @@ public boolean onCreateOptionsMenu(Menu menu) {
     public void onItemClick(TargetLanguage targetLanguage) {
         mSelectedTargetLanguage = targetLanguage;
 
-        // display project list
-        mFragment = new ProjectListFragment();
-        ((ProjectListFragment) mFragment).setArguments(getIntent().getExtras());
-        getFragmentManager().beginTransaction().replace(R.id.fragment_container, (ProjectListFragment) mFragment).commit();
-        // TODO: animate
-        invalidateOptionsMenu();
+        if(!mChangeTargetLanguageOnly) {
+            // display project list
+            mFragment = new ProjectListFragment();
+            ((ProjectListFragment) mFragment).setArguments(getIntent().getExtras());
+            getFragmentManager().beginTransaction().replace(R.id.fragment_container, (ProjectListFragment) mFragment).commit();
+            // TODO: animate
+            invalidateOptionsMenu();
+        } else { // just change the target language
+            Translator translator = App.getTranslator();
+            TargetTranslation targetTranslation = translator.getTargetTranslation(mTargetTranslationId);
+            String targetTranslationId = targetTranslation.getId();
+
+            targetTranslation.changeTargetLanguage(mSelectedTargetLanguage);
+            String newTargetTranslationId = targetTranslation.getId();
+
+            translator.renameTargetTranslation(targetTranslationId, newTargetTranslationId);
```

Comment:
@PhotoNomad0 there's already a method called `normalizePath()` that does this same thing with some extra error checking. I went ahead and switched out the `renameTargetTranslation` method for it.